### PR TITLE
Make the time filtering of the events tables explicit

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

* Add a pinned time filter to events table without changing any behaviour

### the pinned filter

pinned_filter

### querying by time

querying_by_time

(NB there is an existing bug, shown here, where the DateTime picker is ignoring minute selections. That will be solved in a follow-up PR)

## And changes how the eventsTableLogic chooses the after param that filters the events

The events table is limited to the last twelve months of events for performance reasons.

This change means that when:

* fetching events it is always limited to the last twelve months
* polling for events it is limited to the last twelve months when there are no cached events
* polling for events it is limited to since the most recent event when there are cached events
* exporting CSV is is always limited to the last twelve months
* fetching next page of events it is always limited to before the oldest cached event and after one year ago

These changes avoid the possibility of choosing impossible time ranges by setting the after param to dates after those being selected for

## How did you test this code?

running it locally and observing that it has added the pinned filter and that filtering by time behaved as expected

### CODE REVIEW

**Comment by pauldambra** [comment]
Is "Event received > 12 months ago" the best way to communicate we limit to querying only the last twelve months of data? @clarkus

**Comment by pauldambra** [comment]
#8253 for the minute selection bug

**Review by mariusandra — CHANGES_REQUESTED** [review]
- I found time to be annoying about a variable name :)
- Not a big fan of committing in code that's commented out, even if it's a test. Rather use `it.skip` with a clear TODO for how to fix it, or make it work?
- I think it would look cleaner if all filters on this page were grey, not just the pinned one. That's to de-emphasize these blue primary eye-grabbing action targets, which they're not really. 
image
image

- I'd add a little [tooltip](https://ant.design/components/tooltip/) for the pinned filter saying "This filter is pinned and can't be removed" or even explain why you can't remove it. Only seeing the mouse with the "cancel" icon, and not being given an explanation is frightening 👻

File: frontend/src/scenes/events/eventsTableLogic.ts
Comment (pauldambra) [reply frontend/src/scenes/events/eventsTableLogic.ts:229]: I guess this is my RFC to make variable length years 🤣

**Comment by pauldambra** [comment]
All grey and updated with tooltip

Screenshot 2022-01-25 at 14 28 55

**Review by mariusandra — APPROVED** [review]
Guess what, LGTM! 👍 

Since this touches design, I'd pool at least another opinion before merging.

**Comment by pauldambra** [comment]
Yep, I'll wait for @clarkus 👍

**Comment by clarkus** [comment]
> Is "Event received > 12 months ago" the best way to communicate we limit to querying only the last twelve months of data? @clarkus

I think so, but I have a few questions about the problem we're solving. Can users ever clear the pinned filter? Is this something set by the system / product that users can ever change? Can we disallow selection of ranges older than a year? 

Ideally we show this constraint prominently at a level above the filters and disallow the selection of ranges. This would set expectations for users and prevent surprises when they can't select an older value. 

> I think it would look cleaner if all filters on this page were grey, not just the pinned one. That's to de-emphasize these blue primary eye-grabbing action targets, which they're not really.

We should follow the pattern from insights and settings and use the gray pills with the remove action encapsulated within the pill. If that's out of scope, then just going with the gray is fine. Ideally the pills flow left-to-right and fill the horizontal space before stacking. A large volume of filters would displace the events table down the page. Thoughts?

**Comment by pauldambra** [comment]
> I think so, but I have a few questions about the problem we're solving. Can users ever clear the pinned filter? Is this something set by the system / product that users can ever change? Can we disallow selection of ranges older than a year?

The page was silently restricting to 12 months already. Which confused me once I started trying to query by date and time. So we wanted to make it explicit to people

The solved problem is performance

It's not (currently) changeable

We could add restriction on earliest timestamp searched for... I'm moving on to timestamps after this so I'd suggest rolling that change into that work?

(I'd love to fix the crosses being outside of the pills, and the vertical stacking - but hopefully in follow-up to this)

**Comment by clarkus** [comment]
@pauldambra thanks for clarifying. I think a page-level description would be sufficient here. Showing the constraint as a filter just makes it look actionable, and there's nothing the user can do to change that. Could lead to user frustration. The core thing is to make it prominent and then ensure users can't select unsupported ranges. 

Maybe something like
> See events being sent to this project and manage custom actions. Event history is limited to the last twelve months of time.

**Comment by pauldambra** [comment]
updated so it's like this @clarkus

Screenshot 2022-01-25 at 19 50 35

**Comment by clarkus** [comment]
LGTM!

**Comment by Twixes** [comment]
Bad flyby nit: we could avoid the wrap-around if it just said "twelve months", as there aren't really "twelve months of mass" or anything

**Pull request merged**
